app/invocation/invocation_model.tsx

```diff
@@ -426,12 +426,7 @@ export default class InvocationModel {
   }
 
   getEndTimeDate(): Date {
-    let durationMillis = 0;
-    if (!this.finished && this.started) {
-      durationMillis = Math.max(0, new Date().getTime() - this.getStartTimeDate().getTime());
-    } else {
-      durationMillis = Number(this.toolLogMap.get("elapsed time") || 0) * 1_000;
-    }
+    let durationMillis = +this.getDurationMicros() * 1000;
```

Comment:
getDurationMicros() already returns number so the `+` is redundant:

```suggestion
    let durationMillis = this.getDurationMicros() * 1000;
```

Reply:
Done.